### ISSUE TITLE
ISavedGameClient null if Authentication Failed in v0.9.55 and v0.9.54

### ISSUE THREAD

**Issue by TSPetersonD3** [playgameservices/play-games-plugin-for-unity#2387]
I've recently updated my plugin to version 0.9.55. However, I'm having a null reference exception problem.

When using this block here:

```
ISavedGameClient savedGameClient = PlayGamesPlatform.Instance.SavedGame;

savedGameClient.OpenWithAutomaticConflictResolution(_filename, DataSource.ReadCacheOrNetwork, ConflictResolutionStrategy.UseLongestPlaytime, OnSavedGameOpened);
```

savedGameClient is null. 

Obviously, I don't expect it to be able to load any cloud data when the user isn't authenticated, but, it no longer fails as expected. I've also seen this occurs in v0.9.54. However, if I roll back to v0.9.53. It no longer gives me a null ref exception.

I'm not sure if this is intended or not, but previously, if I tried to Open a saved game when a user wasn't authenticated, PlayGamesPlatform.Instance.SavedGame wasn't null and the request would fail with a SavedGameRequestStatus of AuthenticationError.

I've gotten around it by just making a check to see if the user is authenticated, but I just thought I'd point it out in case it's unintended.

**Comment by olehkuznetsov** [comment]
Thank you, seems as a bug.

**Comment by olehkuznetsov** [comment]
Should be fixed in 0.9.56

**Comment by TSPetersonD3** [comment]
Awesome! I'll grab the new version now. Thanks!
